virtcontainers/acrn.go

```diff
@@ -0,0 +1,630 @@
+
+	args := []string{"blkrescan", a.acrnConfig.Name, fmt.Sprintf("%d,%s", slot, drive.File)}
+
+	a.Logger().WithFields(logrus.Fields{
+		"drive": drive,
+		"path":  a.config.HypervisorCtlPath,
+	}).Infof("udpateBlockDevice with acrnctl path")
+	cmd := exec.Command(a.config.HypervisorCtlPath, args...)
+	if err := cmd.Run(); err != nil {
+		a.Logger().WithError(err).Error("updating Block device with newFile path")
+	}
+
+	return err
+}
+
+func (a *acrn) hotplugAddDevice(devInfo interface{}, devType deviceType) (interface{}, error) {
+	span, _ := a.trace("hotplugAddDevice")
+	defer span.Finish()
+
+	switch devType {
+	case blockDev:
+		//The drive placeholder has to exist prior to Update
+		return nil, a.udpateBlockDevice(devInfo.(*config.BlockDrive))
+	default:
+		a.Logger().WithFields(logrus.Fields{"devInfo": devInfo,
+			"deviceType": devType}).Warn("hotplugAddDevice: unsupported device")
+		return nil, fmt.Errorf("hotplugAddDevice: unsupported device: devInfo:%v, deviceType%v",
+			devInfo, devType)
+	}
+}
+
+func (a *acrn) hotplugRemoveDevice(devInfo interface{}, devType deviceType) (interface{}, error) {
+	span, _ := a.trace("hotplugRemoveDevice")
+	defer span.Finish()
+
+	// Not supported. return success
+
+	return nil, nil
+}
+
+func (a *acrn) pauseSandbox() error {
+	span, _ := a.trace("pauseSandbox")
+	defer span.Finish()
+
+	// Not supported. return success
+
+	return nil
+}
+
+func (a *acrn) resumeSandbox() error {
+	span, _ := a.trace("resumeSandbox")
+	defer span.Finish()
+
+	// Not supported. return success
+
+	return nil
+}
+
+// addDevice will add extra devices to Acrn command line.
+func (a *acrn) addDevice(devInfo interface{}, devType deviceType) error {
+	var err error
+	span, _ := a.trace("addDevice")
+	defer span.Finish()
+
+	switch v := devInfo.(type) {
+	case types.Volume:
+		// Not supported. return success
+		err = nil
+	case types.Socket:
+		a.acrnConfig.Devices = a.arch.appendSocket(a.acrnConfig.Devices, v)
+	case kataVSOCK:
+		// Not supported. return success
+		err = nil
+	case Endpoint:
+		a.acrnConfig.Devices = a.arch.appendNetwork(a.acrnConfig.Devices, v)
+	case config.BlockDrive:
+		a.acrnConfig.Devices = a.arch.appendBlockDevice(a.acrnConfig.Devices, v)
+	case config.VhostUserDeviceAttrs:
+		// Not supported. return success
+		err = nil
+	case config.VFIODev:
+		// Not supported. return success
+		err = nil
+	default:
+		break
+	}
+
+	return err
+}
+
+// getSandboxConsole builds the path of the console where we can read
+// logs coming from the sandbox.
+func (a *acrn) getSandboxConsole(id string) (string, error) {
+	span, _ := a.trace("getSandboxConsole")
+	defer span.Finish()
+
+	return utils.BuildSocketPath(store.RunVMStoragePath, id, acrnConsoleSocket)
+}
+
+func (a *acrn) saveSandbox() error {
+	a.Logger().Info("save sandbox")
+
+	// Not supported. return success
+
+	return nil
+}
+
+func (a *acrn) disconnect() {
+	span, _ := a.trace("disconnect")
+	defer span.Finish()
+
+	// Not supported. return success
```

Comment:
Huh? There is no `return` ;)

Reply:
disconnect function doesn't have a return. So removed the misleading comment.